nomenclature/testing.py

```diff
@@ -30,7 +31,7 @@ def assert_valid_yaml(path: Path):
 
 def assert_valid_structure(
     path: Path,
-    dimensions: List[str] = ["region", "variable"],
+    dimensions: List[str] = "[]",
```

Comment:
Same as in `cli.py`, I would avoid `"[]"` as a default. In this special instance it also contradicts the type hint as this is a string and not a list of strings. 
I would go with `None` as the default. For the type hints you can then use `Optional[List[str]]` to indicate to python type checkers that the input can be `None.` Using the same `Optional` for `mappings` might be a good idea as well.